Standards/scs-0219-v1-kaas-networking.md

```diff
@@ -0,0 +1,99 @@
+
+## Design Considerations
+
+The Kubernetes API is arbitrary extensible.
+Many CNI plugins will define custom resources to enable functionality that is not covered in the official [API specification](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/).
+Sometimes they will even reuse names from different API groups, such as `NetworkPolicy`, which exists in the basic `networking.k8s.io/v1` API, but also in `projectcalico.org/v3`.
+
+To avoid any ambiguity, we should therefore be explicit about the API groups and versions of resources.
+We should also avoid mandating third party API extensions, to avoid dependencies on specific third party software and keep the standard as generic as possible.
+
+### Options considered
+
+#### NetworkPolicy API
+
+Kubernetes network policies are used to restrict network traffic between pods in a cluster, but also between pods and external network resources.
+The policy rules can filter based on port and address ranges, but also on Kubernetes-specific target attributes such as namespaces and labels.
+They must be implemented by the CNI plugin, and though they are widely supported, they are still technically optional, and there are some lightweight networking plugins, such as Flannel, that are not enforcing them.
+
+Nonetheless, network policies are widely used and most users will expect them in a managed Kubernetes cluster.
+The wide support among CNI plugins makes them a good target for SCS standardization.
+
+#### Default Network Policies in Namespaces
+
+Basic network policies are namespaced resources, and can only filter traffic to and from pods in their own namespace.
+In a newly created namespace without policies the default behavior will apply, which is to not restrict traffic at all.
+
+It can be desirable to automatically create default network policies in new namespaces, using a policy operator such as Kyverno.
+A CSP could provide such an operator and offer a number of default policies, like blocking connections to other namespaces by default, or blocking access to the OpenStack metadata service.
+
+Any user with permissions to manage their own network policies in a namespace will of course be able to remove or modify any default network policies in that namespace.
+CSP-provided network policies should thus only be viewed as a safety default, and should only be deployed if they are actually beneficial to users.
+
+#### AdminNetworkPolicy API
+
+An alternative to automatically created default network policies are API extensions that allow cluster-wide networking rules.
+Some CNI plugins have implemented such extensions, e.g. Calico's `GlobalNetworkPolicy` and Cilium's `CiliumClusterwideNetworkPolicy`.
+
+The Kubernetes Network Policy Special Interest Group is currently working on an [official API extension](https://network-policy-api.sigs.k8s.io/api-overview/) to cover this functionality.
```

Comment:
```suggestion
The Kubernetes Network Special Interest Group is currently working on an [official API extension](https://network-policy-api.sigs.k8s.io/api-overview/) to cover this functionality.
```

This is the Network Policy working group, which is a subgroup of the Network SIG.
https://network-policy-api.sigs.k8s.io/
https://github.com/kubernetes/community/blob/master/sig-network/README.md

Reply:
sorry, it is the **Network Policy API** working group (with API)